haskell/src/Monocle/Api/Server.hs

```diff
@@ -62,6 +63,30 @@ pattern GetConfig a <- Config.ConfigStatus _ a _
 pattern GetTenants :: [Config.Index] -> Config.ConfigStatus
 pattern GetTenants a <- Config.ConfigStatus _ (Config.Config _ a) _
 
+-- | /login/validate endpoint
+loginLoginValidation ::
+  LoginPB.LoginValidationRequest -> AppM LoginPB.LoginValidationResponse
+loginLoginValidation request = do
+  GetTenants tenants <- getConfig
+  let username = request & LoginPB.loginValidationRequestUsername
+  validated <- traverse (validateOnIndex $ from username) tenants
```

Comment:
I published a playground about how to stop a traverse until a condition is met here: https://github.com/TristanCacqueray/haskell-playground/blob/master/mtl/TraverseMaybeT.hs

Perhaps we could a MaybeT here to avoid going through every workspace?

Reply:
Yes sure that would be more efficient !